R-package/src/Makevars.in

```diff
@@ -0,0 +1,56 @@
+CXX_STD = CXX11
+
+PKGROOT=.
+
+LGB_CPPFLAGS = \
+    @LGB_CPPFLAGS@ \
+    -DUSE_SOCKET \
+    -DLGB_R_BUILD
```

Comment:
Do we need `O3` here for the consistency with `CMakeLists.txt` and `Makevars.win.in`?

Reply:
When I had included `03` before, `R CMD check` complained that that flag is not portable

Reply:
OK, then by default we get `O2` for Linux and macOS.
cc @guolinke 

![image](https://user-images.githubusercontent.com/25141164/88402644-165a5100-cdd4-11ea-95ee-3c4956249ec1.png)


I see that other our dependencies from CRAN are compiled with `O2` as well. So I guess we cannot do anything for that.


But for Windows we have the following:

![image](https://user-images.githubusercontent.com/25141164/88403026-9bde0100-cdd4-11ea-8836-4b29e58ab79a.png)


Our `O3` and default `O2`.

Reply:
yeah it is one of the quirks of CRAN that there are different rules for Windows vs. non-Windows. So I was able to include 03 in `R-package/src/Makevars.win.in` to more closely match our `CMakeLists`: https://github.com/jameslamb/LightGBM/blob/feat/cran-install/R-package/src/Makevars.win.in#L9

That extra `02` in the Windows build is something R has chosen to set itself, not something I set (you won't see it directly in https://github.com/jameslamb/LightGBM/blob/feat/cran-install/R-package/src/Makevars.win.in or in https://github.com/jameslamb/LightGBM/blob/feat/cran-install/R-package/configure.win)

Reply:
> Our `O3` and default `O2`.

Unfortunately, we still get `O2` actually due to the order.
> If you use multiple -O options, with or without level numbers, the last such option is the one that is effective. 
    https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html

Given this fact, I think we can remove yseless `O3` from `Makevars.win.in` to avoid the confusion.

Reply:
ohhhh I didn't know that. Ok I'll just remove it

Reply:
removed: https://github.com/microsoft/LightGBM/pull/3188/commits/8703a9080725281b824a12b77862ec6eab48a36d